### PR TITLE
Simple test Parent Child

### PR DESCRIPTION
**Pull request by BogdanQA**
User story https://trello.com/c/MAXbweUC

### CODE REVIEW

**Review by guram21 — APPROVED** [review]

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        WebElement inputDecimal = driver.findElement(By.xpath("//input[@id='decimal']"));
+        inputDecimal.sendKeys(DECIMAL);
+
+        WebElement inputDate = driver.findElement(By.xpath("//input[@id='date']"));
+        inputDate.sendKeys(DATE);
+
+        WebElement inputDateTime = driver.findElement(By.xpath("//input[@id='datetime']"));
+        inputDateTime.sendKeys(DATETIME);
+
+        Select Dropdown = new Select(driver.findElement(By.xpath("//select[@name='entity_form_data[user]']")));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:55]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        WebElement inputDecimal = driver.findElement(By.xpath("//input[@id='decimal']"));
+        inputDecimal.sendKeys(DECIMAL);
+
+        WebElement inputDate = driver.findElement(By.xpath("//input[@id='date']"));
+        ProjectUtils.fill(getWebDriverWait(),inputDate, DATE);
+
+        WebElement inputDateTime = driver.findElement(By.xpath("//input[@id='datetime']"));
+        ProjectUtils.fill(getWebDriverWait(),inputDateTime, DATETIME);
+
+        Select Dropdown = new Select(driver.findElement(By.xpath("//select[@name='entity_form_data[user]']")));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:97]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:145]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        inputDecimal.sendKeys(DECIMAL);
+
+        WebElement inputDate = driver.findElement(By.xpath("//input[@id='date']"));
+        inputDate.sendKeys(DATE);
+
+        WebElement inputDateTime = driver.findElement(By.xpath("//input[@id='datetime']"));
+        inputDateTime.sendKeys(DATETIME);
+
+        Select Dropdown = new Select(driver.findElement(By.xpath("//select[@name='entity_form_data[user]']")));
+        Dropdown.selectByVisibleText("user132@tester.com");
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:56]: "user132@tester.com" used multiple times, you can save this string in variable

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        ProjectUtils.fill(getWebDriverWait(),inputDateTime, DATETIME);
+
+        Select Dropdown = new Select(driver.findElement(By.xpath("//select[@name='entity_form_data[user]']")));
+        Dropdown.selectByVisibleText("user132@tester.com");
+
+        WebElement saveButton = driver.findElement(By.xpath("//button[@id='pa-entity-form-save-btn']"));
+        saveButton.click();
+
+        List<WebElement> childRecord = driver.findElements(By.xpath("//tbody/tr[1]"));
+        Assert.assertEquals(childRecord.size(), 1);
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:104]: only one element can be found with xpath in line 103, but test should find elements for all records
replace //tbody/tr[1] with //tbody/tr

**Review by Stan575 — CHANGES_REQUESTED** [review]

File: src/test/java/test/entity/EntityParentChildViewTest.java (hunk 5 above)
Comment (BogdanQA) [reply src/test/java/test/entity/EntityParentChildViewTest.java]: Thanks! Fixed

File: src/test/java/test/entity/EntityParentChildViewTest.java (hunk 4 above)
Comment (BogdanQA) [reply src/test/java/test/entity/EntityParentChildViewTest.java]: Thank you! Fixed

Comment (BogdanQA) [reply src/test/java/test/entity/EntityParentChildViewTest.java]: Fixed

File: src/test/java/test/entity/EntityParentChildViewTest.java (hunk 3 above)
Comment (BogdanQA) [reply src/test/java/test/entity/EntityParentChildViewTest.java:146]: Fixed

File: src/test/java/test/entity/EntityParentChildViewTest.java (hunk 2 above)
Comment (BogdanQA) [reply src/test/java/test/entity/EntityParentChildViewTest.java:98]: Thank you! Fixed

File: src/test/java/test/entity/EntityParentChildViewTest.java (hunk 1 above)
Comment (BogdanQA) [reply src/test/java/test/entity/EntityParentChildViewTest.java:56]: Fixed

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+        WebDriver driver = getDriver();
+
+        WebElement parentMenu = driver.findElement(By.xpath("//p[contains(text(),'Parent')]"));
+        parentMenu.click();
+
+        WebElement createNewParent = driver.findElement(By.xpath("//i[contains(text(),'create_new_folder')]"));
+        createNewParent.click();
+
+        WebElement InputString = driver.findElement(By.xpath("//input[@id='string']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:38]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+        InputString.sendKeys(STRING);
+
+        WebElement InputText = driver.findElement(By.xpath("//textarea[@id='text']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:41]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+        InputText.sendKeys(TEXT);
+
+        WebElement InputInt = driver.findElement(By.xpath("//input[@id='int']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:44]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+        InputInt.sendKeys(INT);
+
+        WebElement InputDecimal = driver.findElement(By.xpath("//input[@id='decimal']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:47]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+        InputDecimal.sendKeys(DECIMAL);
+
+        WebElement InputDate = driver.findElement(By.xpath("//input[@id='date']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:50]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+        InputDate.sendKeys(DATE);
+
+        WebElement InputDateTime = driver.findElement(By.xpath("//input[@id='datetime']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:53]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+        WebElement parentMenu = driver.findElement(By.xpath("//p[contains(text(),'Parent')]"));
+        parentMenu.click();
+
+        WebElement parentRecord = driver.findElement(By.xpath("//tbody/tr[1]"));
+        parentRecord.click();
+
+        WebElement createNewChild = driver.findElement(By.xpath("//i[contains(text(),'create_new_folder')]"));
+        createNewChild.click();
+
+        WebElement InputString = driver.findElement(By.xpath("//input[@id='string']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:80]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+        InputString.sendKeys(STRING);
+
+        WebElement InputText = driver.findElement(By.xpath("//textarea[@id='text']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:83]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+        InputText.sendKeys(TEXT);
+
+        WebElement InputInt = driver.findElement(By.xpath("//input[@id='int']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:86]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+        InputInt.sendKeys(INT);
+
+        WebElement InputDecimal = driver.findElement(By.xpath("//input[@id='decimal']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:89]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+        InputDecimal.sendKeys(DECIMAL);
+
+        WebElement InputDate = driver.findElement(By.xpath("//input[@id='date']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:92]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+        ProjectUtils.fill(getWebDriverWait(),InputDate, DATE);
+
+        WebElement InputDateTime = driver.findElement(By.xpath("//input[@id='datetime']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:95]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+        WebElement parentMenu = driver.findElement(By.xpath("//p[contains(text(),'Parent')]"));
+        parentMenu.click();
+
+        WebElement parentRecord = driver.findElement(By.xpath("//tbody/tr[1]"));
+        parentRecord.click();
+
+        WebElement createNewChild = driver.findElement(By.xpath("//i[contains(text(),'create_new_folder')]"));
+        createNewChild.click();
+
+        WebElement InputString = driver.findElement(By.xpath("//input[@id='string']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:128]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+        InputString.sendKeys(STRING);
+
+        WebElement InputText = driver.findElement(By.xpath("//textarea[@id='text']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:131]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+        InputText.sendKeys(TEXT);
+
+        WebElement InputInt = driver.findElement(By.xpath("//input[@id='int']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:134]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 21]
```diff
@@ -0,0 +1,162 @@
...
+        InputInt.sendKeys(INT);
+
+        WebElement InputDecimal = driver.findElement(By.xpath("//input[@id='decimal']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:137]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 22]
```diff
@@ -0,0 +1,162 @@
...
+        InputDecimal.sendKeys(DECIMAL);
+
+        WebElement InputDate = driver.findElement(By.xpath("//input[@id='date']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:140]: variable name starts with capital letter

File: src/test/java/test/entity/EntityParentChildViewTest.java [hunk 23]
```diff
@@ -0,0 +1,162 @@
...
+        ProjectUtils.fill(getWebDriverWait(),InputDate, DATE);
+
+        WebElement InputDateTime = driver.findElement(By.xpath("//input[@id='datetime']"));
```
Comment (Stan575) [inline src/test/java/test/entity/EntityParentChildViewTest.java:143]: variable name starts with capital letter

File: src/test/java/test/portal/MarketplaceInstanceTest.java [hunk 24]
```diff
@@ -36,7 +36,7 @@
     private static final By RESET = By.xpath("//a[contains(text(), 'Reset')]");
     private static final By CONGRATS_TEXT = By.xpath("//div[@class='card-body ']/child::div/child::h3[1]");
     private static final String PRIMARY_LANGUAGE = "English";
-    private static final List<String> ascNames = Arrays.asList("aaaa", "bbbb");
+    private static final List<String> ascNames = Arrays.asList("aaaaa", "bbbbb");
```
Comment (Stan575) [inline src/test/java/test/portal/MarketplaceInstanceTest.java:39]: are this changes related to https://trello.com/c/MAXbweUC/525-entity-parent-child-view ?

**Review by Stan575 — CHANGES_REQUESTED** [review]

File: src/test/java/test/portal/MarketplaceInstanceTest.java (hunk 24 above)
Comment (BogdanQA) [reply src/test/java/test/portal/MarketplaceInstanceTest.java:39]: no its MarketplaceInstanceTest

**Comment by Stan575** [comment]
@BogdanQA sorry for not being explicit in comments

File: src/test/java/test/portal/MarketplaceInstanceTest.java (hunk 24 above)
Comment (Stan575) [reply src/test/java/test/portal/MarketplaceInstanceTest.java:39]: I think it should be included into another PR

Comment (BogdanQA) [reply src/test/java/test/portal/MarketplaceInstanceTest.java:39]: OK

**Pull request closed without merging**